web3/contract.py

```diff
@@ -1339,7 +1350,7 @@ def __getattr__(self, function_name: str) -> Any:
                 "The ABI for this contract contains no function definitions. ",
                 "Are you sure you provided the correct contract ABI?"
             )
-        elif function_name not in self._functions:
+        elif function_name not in [fn['name'] for fn in self._functions]:
```

Comment:
I've tried to not change any source code while type hinting, but this one to me seems like a genuine bug

Reply:
I'd ask that you extract this to a stand-alone PR and write a test that would have exposed this bug.

Reply:
Also, when doing inclusion checks you should default to use a `set()`